packages/dataviews/src/dataviews-layouts/index.ts

```diff
@@ -30,6 +30,7 @@ export const VIEW_LAYOUTS = [
 		label: __( 'Grid' ),
 		component: ViewGrid,
 		icon: category,
+		supportsDensity: true,
```

Comment:
This could also be an array of supported presets.

Reply:
We already have some hardcode logic in the view config that is layout specific. Note how the arrows to "move fields right/left" only show up in the table layout is active but not in any other layout:

<img width="366" alt="Screenshot 2024-11-13 at 10 52 14" src="https://github.com/user-attachments/assets/163f0b0f-feb3-4804-ac73-f98b831dce4f">

This is [hardcoded logic](https://github.com/WordPress/gutenberg/blob/0087c4f5e7efb33db1d38b08c0f70f0de5f60b2c/packages/dataviews/src/components/dataviews-view-config/index.tsx#L452) we should aim to remove. The view config components should be layout agnostic, otherwise, we cannot open it to 3rd party layouts.

Connecting that thought with what you're doing here, it may be a good time to design this API to "declare features supported by the layout" instead of making it specific to density. For example, using `supports: [ 'layout' ]` or `supports: { layout: true }` instead of `supportsDensity: true`.

Reply:
I agree and was something I had in mind to do (your suggestion) if we had another case in the future. Since we have, I'll do it now. Thanks!